server/src/server.ts

```diff
@@ -165,22 +162,51 @@ function logErrors<T>(f: () => T): T {
   }
 }
 
-connection.onDefinition((textDocumentPosition: TextDocumentPositionParams): Definition => logErrors(() => {
-  const {fileName, service, offset, languageId} = documents.getServiceInfo(textDocumentPosition.textDocument,
-    textDocumentPosition.position)
+connection.onDefinition((params: lsp.TextDocumentPositionParams) => logErrors(() => {
+  const {textDocument, position} = params;
+  const {fileName, service, offset} = documents.getServiceInfo(textDocument, position);
   if (fileName && service && offset != null) {
     let result = service.getDefinitionAt(fileName, offset);
-    if (result) {
-      return ngDefintionToDefintion(result);
+    if (!result) {
+      return;
+    }
+    if (Array.isArray(result)) {
+      // Backwards compatibility with old ng.Location[] (ng.Definition)
+      return ngDefinitionToDefinition(result);
     }
+    const {textSpan, definitions} = result as ts.DefinitionInfoAndBoundSpan;
+    if (!definitions || !definitions.length) {
+      return;
+    }
+    const [start, end] = documents.offsetsToPositions(textDocument, [
+      textSpan.start,
+      textSpan.start + textSpan.length,
+    ]);
+    const originSelectionRange = lsp.Range.create(
+      start.line - 1, start.character - 1, end.line - 1, end.character - 1);
```

Comment:
@kyliau, you don't need to adjust line and character position. It already have correct position.

Reply:
Isn't vscode 1-based whereas LSP is 0-based?

Reply:
I don't know. I only tested changes from angular/angular#32098 and angular/angular#32085. I saw these bugs: 
"Go To Definition" doesn't work for html tags.
"Go To Definition" navigates to wrong line and position for html attributes.
 It work as expected when I remove " -1" from 186 and 194 lines

Reply:
thanks, I'll double check this.